core/src/main/java/nu/yona/server/analysis/entities/DayActivity.java

```diff
@@ -34,7 +35,11 @@ public static DayActivityRepository getRepository()
 		return (DayActivityRepository) RepositoryProvider.getRepository(DayActivity.class, UUID.class);
 	}
 
-	@OneToMany(cascade = CascadeType.ALL)
+	// because we are caching DayActivity instances and using this list, the fetch type has to be eager
+	// (or we should re-initialize a Hibernate session after returning from the cache)
+	// see http://jira.yona.nu/browse/YD-352 and
+	// http://stackoverflow.com/questions/26507446/how-to-resolve-lazyinitializationexception-in-spring-data-jpa
+	@OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
```

Comment:
How certain are you that this works? I have tried this approach earlier and failed. That was inexplicable to me, but still...
Can we have a test case that reproduces the issue?